### PR TITLE
New Recipe: Cybersecurity Firewall analysis with Splunk

### LINKED ISSUE #104: Cyber Security: Identify Firewall Deny events from Splunk
**Describe the use case**
The purpose of this project is to provide a demonstration on how to optimize your Splunk data ingestion by using Confluent. This guide will route data from a Splunk Universal Forwarder running an eventgenerator, to the Confluent Splunk-2-Splunk Source Connector while bypassing the Splunk indexing layer and retaining all of the critical metadata assosciated with each data source (source, sourcetype, host, event, _meta).

siem_optimization

**Provide the ksqlDB application**
Document the full set of ksqlDB statements that represent the use case.

https://github.com/JohnnyMirza/confluent_splunk_demo/blob/main/statements.sql

### PR DESCRIPTION
**Pull request by ybyzek**
Addresses https://github.com/confluentinc/ksqldb-recipes/issues/104

### CODE REVIEW

**Comment by bbejeck** [comment]
I started to review and when I run
```
docker build \
   -t localbuild/connect_distributed_with_splunk-s2s:1.0.5 \
   -f Dockerfile .
```
I'm getting this error

```
=> [internal] load build definition from Dockerfile                                                                                                                     0.0s
 => => transferring dockerfile: 489B                                                                                                                                     0.0s
 => [internal] load .dockerignore                                                                                                                                        0.0s
 => => transferring context: 2B                                                                                                                                          0.0s
 => [internal] load metadata for docker.io/confluentinc/cp-server-connect-base:6.2.0                                                                                     0.6s
 => [auth] confluentinc/cp-server-connect-base:pull token for registry-1.docker.io                                                                                       0.0s
 => [internal] load build context                                                                                                                                        0.0s
 => => transferring context: 2B                                                                                                                                          0.0s
 => CANCELED [1/3] FROM docker.io/confluentinc/cp-server-connect-base:6.2.0@sha256:103e172d70e1da08b84f0c2fb1d39ad4cf5c2eb69a7763fe54504273a61b2db2                      0.0s
 => => resolve docker.io/confluentinc/cp-server-connect-base:6.2.0@sha256:103e172d70e1da08b84f0c2fb1d39ad4cf5c2eb69a7763fe54504273a61b2db2                               0.0s
 => => sha256:103e172d70e1da08b84f0c2fb1d39ad4cf5c2eb69a7763fe54504273a61b2db2 3.05kB / 3.05kB                                                                           0.0s
 => => sha256:1e49a220dc1cd261a8af9253bd91422f6029420b7649e1de93f052e72bedd1b0 24.93kB / 24.93kB                                                                         0.0s
 => ERROR [2/3] COPY --chown=appuser:appuser include/etc/confluent/docker /etc/confluent/docker                                                                          0.0s
------
 > [2/3] COPY --chown=appuser:appuser include/etc/confluent/docker /etc/confluent/docker:
------
failed to compute cache key: "/include/etc/confluent/docker" not found: not found
```

**Comment by ybyzek** [comment]
@bbejeck Dockerfile fixed in latest commit -- can you please try again?

**Review by bbejeck — APPROVED** [review]
Nice addition @ybyzek  LGTM 
Validated on Confluent Cloud

File: docs/cybersecurity/firewall-splunk/index.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+### Set up your environment
+
+Provision a Kafka cluster in [Confluent Cloud](https://www.confluent.io/confluent-cloud/tryfree/?utm_source=github&utm_medium=ksqldb_recipes&utm_campaign=firewall).
+
+--8<-- "docs/shared/ccloud_setup.md"
+
+### Read the data in
+
+--8<-- "docs/shared/self_managed_connector.md"
```
[the remark this answers is not in the document]
Comment (ybyzek) [reply docs/cybersecurity/firewall-splunk/index.md:27]: The particular connector used in this recipe is not available in Confluent Cloud.  (It's unrelated to ksqlDB-connect integration)

File: docs/cybersecurity/firewall-splunk/process.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  event VARCHAR,
+  time BIGINT,
+  host VARCHAR,
+  source VARCHAR,
+  index VARCHAR,
+  sourcetype VARCHAR
+) WITH (
+  KAFKA_TOPIC = 'splunk-s2s-events',
+  VALUE_FORMAT = 'json',
+  PARTITIONS = 6
```
[the remark this answers is not in the document]
Comment (ybyzek) [reply docs/cybersecurity/firewall-splunk/process.sql:13]: @mjsax If a user does not have a connector and just wants to use `INSERT INTO` (which I believe will be a large number of users running recipes), then `PARTITIONS` is required.

File: docs/cybersecurity/firewall-splunk/process.sql [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+);
+
+-- Split message into array
+CREATE STREAM splunk_parsed WITH (KAFKA_TOPIC = 'splunk_parsed') AS SELECT
+  *,
```
[the remark this answers is not in the document]
Comment (ybyzek) [reply docs/cybersecurity/firewall-splunk/process.sql]: @mjsax agree, thanks for the suggestion.  Addressed in latest commit

File: docs/cybersecurity/firewall-splunk/process.sql [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  host,
+  sourcetype,
+  KSQL_COL_0[2] AS action,
+  KSQL_COL_0[3] AS protocol,
+  SPLIT(REGEXP_SPLIT_TO_ARRAY(KSQL_COL_0[5], 'inside:')[2], '/')[1] as inside_ip,
+  SPLIT(REGEXP_SPLIT_TO_ARRAY(KSQL_COL_0[5], 'inside:')[2], '/')[2] as inside_port,
+  SPLIT(REGEXP_SPLIT_TO_ARRAY(KSQL_COL_0[7], 'outside:')[2], '/')[1] as outside_ip,
+  SPLIT(REGEXP_SPLIT_TO_ARRAY(KSQL_COL_0[7], 'outside:')[2], '/')[2] as outside_port
+FROM splunk_parsed
+WHERE sourcetype = 'cisco:asa' and KSQL_COL_0[2] = 'Deny'
```
[the remark this answers is not in the document]
Comment (ybyzek) [reply docs/cybersecurity/firewall-splunk/process.sql]: @mjsax agree, thanks for the suggestion.  Addressed in latest commit

File: mkdocs.yml [hunk 5]
```diff
@@ -18,6 +18,7 @@ nav:
     - Build customer loyalty programs: customer-360/loyalty-rewards/index.md
   - Cybersecurity:
     - Detect and analyze SSH attacks: cybersecurity/SSH-attack/index.md
+    - Identify firewall deny events from Splunk: cybersecurity/firewall-splunk/index.md
```
Comment (ybyzek) [inline mkdocs.yml:21]:
```suggestion
    #- Identify firewall deny events from Splunk: cybersecurity/firewall-splunk/index.md
```

**Pull request merged**